src/main/java/hexgui/gui/GuiPreferences.java

```diff
@@ -0,0 +1,78 @@
+// ----------------------------------------------------------------------------
+// $Id$
+// ----------------------------------------------------------------------------
+
+package hexgui.gui;
+
+import java.io.*;
+import java.util.*;
+import java.util.prefs.*;
+
+// ----------------------------------------------------------------------------
+
+/** GuiPreferences */
+public final class GuiPreferences {
+  private static String[][] s_preflist =
+      new String[][] {
+        {"gui-board-type", "Flat"},
+        {"gui-board-orientation", "positive"},
+        {"gui-board-pixel-width", "750"},
+        {"gui-board-pixel-height", "500"},
+        {"gui-board-width", "11"},
+        {"gui-board-height", "11"},
+        {"draw-field-alpha", "0.3"}, // FIXME: not used yet!
+        {"gui-toolbar-visible", "true"},
+        {"shell-show-on-connect", "false"},
+        {"analyze-show-on-connect", "false"},
+        {"auto-respond", "true"},
+        {"first-move-color", "black"},
+        {"remote-host-name", "localhost"},
+        {"is-program-attached", "false"},
+        {"attached-program", "dummy-program-name"},
+        {"path-load-game", "."},
+        {"path-save-game", "."},
+        {"dummy-preference", ""}
+      };
+
+  public GuiPreferences(Class theClass) {
+    m_preferences = Preferences.userNodeForPackage(theClass);
+  }
+
+  public String get(String name) {
+    for (int i = 0; i < s_preflist.length; i++) {
+      if (name.equals(s_preflist[i][0])) {
+        return m_preferences.get(s_preflist[i][0], s_preflist[i][1]);
+      }
+    }
+    System.out.println("Unknown preference: " + name + ", returning 'unknown'.");
+    return "unknown";
+  }
+
+  public int getInt(String name) {
+    return Integer.parseInt(get(name));
+  }
+
+  public boolean getBoolean(String name) {
+    return Boolean.parseBoolean(get(name));
+  }
+
+  public void put(String name, String value) {
+    for (int i = 0; i < s_preflist.length; i++) {
+      if (name.equals(s_preflist[i][0])) {
+        m_preferences.put(s_preflist[i][0], value);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem:**
 This line of code lacks validation when processing input data through the following parameter: &#39;value&#39; (index: 1 | type: String). The parameter is exposed to external callers, because its enclosing class and method are publicly accessible. This means that upstream validation, if it exists, can be bypassed. Other validated parameters: &#39;name&#39;. Malicious, malformed, or unbounded inputs can cause unexpected runtime behavior or crashes, and can slow performance.

**Fix:**
 Add checks to ensure the validity of the parameter&#39;s value, such as testing it for nullness (for example, using the @NonNull annotation as [described in the Lombok library](https://projectlombok.org/features/NonNull)), emptiness, or equality. Or to prevent direct calls to it, [reduce the method&#39;s visibility](https://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html).

 Learn more about potential threats and guidance from the [Common Weakness Enumeration website](https://cwe.mitre.org/data/definitions/693.html) and the [OWASP Cheat Sheet series](https://cheatsheetseries.owasp.org/cheatsheets/Input_Validation_Cheat_Sheet.html).